scipy/optimize/_differentialevolution.py

```diff
@@ -469,6 +498,40 @@ def init_population_random(self):
         # reset number of function evaluations counter
         self._nfev = 0
 
+    def init_population_array(self, init):
+        """
+        Initialises the population with a user specified population.
+
+        Parameters
+        ----------
+        init : np.ndarray
+            Array specifying subset of the initial population. The array should
+            have shape (M, len(x)), where len(x) is the number of parameters.
+            The population is clipped to the lower and upper `bounds`.
+        """
+        # make sure you're using a float array
+        popn = np.asfarray(init)
+
+        if (np.size(popn, 0) < 5 or
+                    popn.shape[1] != self.parameter_count or
+                    len(popn.shape) != 2):
+            raise ValueError(self.__init_error_msg)
+
+        # scale values and clip to bounds, assigning to population
+        self.population = np.clip(self._unscale_parameters(popn), 0, 1)
+
+        self.num_population_members = np.size(self.population, 0)
+
+        self.population_shape = (self.num_population_members,
+                                 self.parameter_count)
+
+        # reset population energies
+        self.population_energies = (np.ones(self.num_population_members) *
+                                    np.inf)
+
+        # reset number of function evaluations counter
+        self._nfev = 0
```

Comment:
There's some copy/paste of other `init_population_*` methods here. Not a blocker for this change, but could be worth refactoring a bit.